### PR TITLE
Fix incorrect IPv6 interface index on Windows

### PR DESCRIPTION
**Pull request by rolznz**

### CODE REVIEW

File: src/mdns.c [hunk 1]
```diff
@@ -208,7 +208,7 @@ mdns_list_interfaces(multicast_if** pp_intfs, size_t* p_nb_intf, int ai_family)
                 if (!mdns_is_interface_valuable(current))
                         continue;
                 if (ai_family == AF_INET6)
-                        *intfs = htonl(current->Ipv6IfIndex);
+                        *intfs = current->Ipv6IfIndex;
                 else
                         *intfs = htonl(current->IfIndex);
```
Comment (tguillem) [inline src/mdns.c:220]: You should also remove htonl here, no ?

Comment (rolznz) [reply src/mdns.c:220]: I thought so too, but when I tried libmicrodns would no longer init correctly. It seems the EDIT: interface index is handled differently in Windows for v4/v6.

Comment (rolznz) [reply src/mdns.c:220]: @tguillem please look at these two:

https://docs.microsoft.com/en-us/windows/win32/winsock/ipproto-ipv6-socket-options

IPV6_MULTICAST_IF

The input value for setting this option is a 4-byte interface index of the desired outgoing interface in **host byte order**.

https://docs.microsoft.com/en-us/windows/win32/winsock/ipproto-ip-socket-options

IP_MULTICAST_IF

The input value for setting this option is a 4-byte IPv4 address in **network byte order**.

Would you like me to add a comment to the commit?

Comment (tguillem) [reply src/mdns.c:220]: Good catch, I searched for such information but could not find it. Yes a comment will be really appreciated.

Comment (rolznz) [reply src/mdns.c:220]: @tguillem Thanks - done.

**Pull request merged**